src/gui/sdlmain.cpp

```diff
@@ -3307,16 +3196,44 @@ static SDL_Rect calc_viewport_fit(const int win_w, const int win_h)
 	// limit the window to the user's desired viewport, if configured
 	const auto [lwin_w, lwin_h] = restrict_to_viewport_resolution(win_w, win_h);
 
-	// calculate the aspect ratios of the draw buffer and the limited window
+	// calculate the aspect ratio of the draw buffer
 	const auto draw_aspect = (sdl.draw.width * sdl.draw.scalex) /
 	                         (sdl.draw.height * sdl.draw.scaley);
-	const auto lwin_aspect = static_cast<double>(lwin_w) / lwin_h;
 
-	// calculate the viewport contingent on the aspect ratio of the limited
-	// window versus draw buffer
-	const auto lwin_is_wider = lwin_aspect > draw_aspect;
-	const auto view_w = lwin_is_wider ? iround(lwin_h * draw_aspect) : lwin_w;
-	const auto view_h = lwin_is_wider ? lwin_h : iround(lwin_w / draw_aspect);
+	int view_w = 0;
```

Comment:
Could we turn this function into something that can be easily comprehended by an average person unfamiliar with the code in a single pass?

I find the variable names confusing and inconsistent. The problem starts here:

```
const auto [lwin_w, lwin_h] = restrict_to_viewport_resolution(win_w, win_h);
```

Why `lwin`? Can we find a better name the describes this is the maximum bounds of the viewport we want to fit our image into? (hint: `bound_w`, `bound_h`).

Reply:
Similarly, `draw_aspect` is very ambiguous to me... "Draw" can be anything; we're "drawing" here, right? 😅 

Can we be more precise and call this what it is? `source_aspect` would be a lot better I think. But technically, this is the _storage aspect ratio_, maybe call it that?